### PR TITLE
Updated the unit test for BlobstoreIterator into a table format

### PR DESCRIPTION
**Pull request by agautam478**
**What changed?**

Converted the already present test that was using Testing suite into and independent table test.
- The `pItr` variable is an iterator created by fetcher.ConcreteExecutionIterator, which iterates over concrete executions.
- The `NewBlobstoreWriter` is used to create a writer for the blobstore, and executions are added to this writer.
- After adding all executions, the writer is flushed, and assertions are made to check the correctness of the operation.
- The `NewBlobstoreIterator` is then used to iterate over the written blobs, and assertions are made to ensure the data integrity.

**Why?**
These changes were made as a part of our initiative to increase the code coverage and also to maintain consistency with it's sister `BlobstoreWriter` test.

**How did you test it?**
yes. Tested locally.

**Potential risks**
NA

**Release notes**
NA

**Documentation Changes**
NA

### CODE REVIEW

File: common/reconciliation/store/writerIterator_test.go [hunk 1]
```diff
@@ -48,60 +46,60 @@ var (
...
-func (s *WriterIteratorSuite) SetupTest() {
-	s.Assertions = require.New(s.T())
-}
+func TestWriterIterator(t *testing.T) {
+	testCases := []struct {
+		name         string
+		pages        int
+		countPerPage int
+	}{
+		{"StandardCase", 10, 10},
```
Comment (taylanisikdemir) [inline common/reconciliation/store/writerIterator_test.go:55]: are there different error cases to cover with new test cases?

File: service/history/handler.go [hunk 2]
```diff
@@ -26,6 +26,7 @@ import (
 	"context"
 	"errors"
 	"fmt"
+	"github.com/uber/cadence/service/history/workflow"
```
Comment (taylanisikdemir) [inline service/history/handler.go:29]: is this added by mistake?

Comment (agautam478) [reply service/history/handler.go:29]: I am not sure where that came from.

File: common/reconciliation/store/writerIterator_test.go (hunk 1 above)
Comment (taylanisikdemir) [reply common/reconciliation/store/writerIterator_test.go:55]: yeah let's add the ones that validates pagination works

**Review by taylanisikdemir — APPROVED** [review]

**Pull request merged**